modules/ui/maproulette_details.js

```diff
@@ -0,0 +1,319 @@
+            const valuesStart =
+                segment.indexOf('values=&quot;') + 'values=&quot;'.length;
+            const valuesEnd = segment.indexOf('&quot;', valuesStart);
+            const options = segment
+                .substring(valuesStart, valuesEnd)
+                .split(',');
+            const dropdownHtml = generateDropdownHtml(dropdownName, options);
+            const remainder = segment
+                .substring(valuesEnd + '&quot;'.length)
+                .trim()
+                .replace(/^\]/, '');
+            transformedText += dropdownHtml + remainder;
+        });
+        return transformedText;
+    }
+
+    function replaceMustacheTags(text, task) {
+        if (!text) return '';
+        const tagRegex = /\{\{([\w:]+)\}\}/g;
+
+        // Build a map of all properties similar to Rapid's implementation
+        function buildAllProperties(obj) {
+            const all = new Map();
+            if (!obj) return all;
+            // Preferred: task.taskFeatures (array of features with .properties)
+            if (Array.isArray(obj.taskFeatures)) {
+                obj.taskFeatures
+                    .map((f) => (f && f.properties) || {})
+                    .forEach((props) => {
+                        Object.keys(props).forEach((key) =>
+                            all.set(key, props[key]),
+                        );
+                    });
+            }
+            // Fallback common shapes
+            if (obj.properties) {
+                Object.keys(obj.properties).forEach((key) =>
+                    all.set(key, obj.properties[key]),
+                );
+            }
+            const geom = obj.geometries || obj.geojson || obj.geometry;
+            if (geom) {
+                if (geom.properties) {
+                    Object.keys(geom.properties).forEach((key) =>
+                        all.set(key, geom.properties[key]),
+                    );
+                }
+                if (Array.isArray(geom.features) && geom.features.length) {
+                    const featProps =
+                        geom.features[0] && geom.features[0].properties;
+                    if (featProps) {
+                        Object.keys(featProps).forEach((key) =>
+                            all.set(key, featProps[key]),
+                        );
+                    }
+                }
+            }
+            // Flat fields
+            Object.keys(obj).forEach((key) => {
+                if (!all.has(key)) all.set(key, obj[key]);
+            });
+            return all;
+        }
+
+        const allProps = buildAllProperties(task);
+
+        return text.replace(tagRegex, (match, propertyName) => {
+            if (propertyName === 'osmIdentifier' && task && task.title) {
+                const osmId = String(task.title).split('@')[0];
+                return `<a href="#" class="highlight-link" data-osm-id="${osmId}">${osmId}</a>`;
```

Comment:
This part likely does not work, yet. In Rapid it allows to hover this link to highlight the referenced feature.